Dockerfile

```diff
@@ -22,9 +22,9 @@ WORKDIR web-apps
 # Install JavaScript dependencies and manually check if yarn.lock needs an update.
 # Yarn v1 doesn't respect the --frozen-lockfile flag when using workspaces.
 # https://github.com/yarnpkg/yarn/issues/4098
-RUN sha384sum yarn.lock > yarn-lock-sha \
+RUN cp yarn.lock yarn-before-install.lock \
   && yarn install \
-  && sha384sum --check yarn-lock-sha || \
+  && git diff --no-index --exit-code yarn-before-install.lock yarn.lock || \
```

Comment:
`diff` could be used here rather than `git diff` but

1) git is already available
2) most devs are probably more familiar with the output of `git diff` rather than `diff`
3) `git diff` has arguably more useful output than `diff`: it shows context and file names